client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -878,7 +881,17 @@ function mediaButton( editor ) {
 	editor.on( 'touchstart touchmove touchend', selectImageOnTap() );
 
 	editor.on( 'init', function () {
-		MediaStore.on( 'change', updateMedia );
+		unsubscribeFromReduxStore = store.subscribe( () => {
+			const { ID: siteId } = getSelectedSite( store.getState() );
+			if ( ! siteId ) {
+				return;
+			}
+			previousMediaReference = currentMediaReference;
+			currentMediaReference = getMedia( store.getState(), siteId );
+			if ( ! isEqual( previousMediaReference, currentMediaReference ) ) {
```

Comment:
To detect any change in the immutable Redux state tree, a `===` comparison is usually enough. Do we really need to use `isEqual` here?

Reply:
I have to dig further to find out why but comparing both approaches, using `previousMediaReference !== currentMediaReference` the `updateMedia` function is called 17 times vs 2 times when using `isEqual`.

Reply:
It would be interesting to know where these no-op changes are coming from. Usually, an action that doesn't _really_ change a piece of state should leave the actual object unmodified.

Is it some inefficient reducer? Or is there a selector that doesn't do memoization properly?

Reply:
After a bit of digging, it seems that the [QueryManager.getItems](https://github.com/Automattic/wp-calypso/blob/master/client/lib/query-manager/index.js#L160) method, used by the `getMedia` selector, is to blame. It creates a new array instance on every call.

Reply:
Trying to add memoization here: #44986

Reply:
Resolved with f3a7d18